Model/ProductProtection.php

```diff
@@ -0,0 +1,87 @@
+<?php
+/*
+ * Copyright Extend (c) 2023. All rights reserved.
+ * See Extend-COPYING.txt for license details.
+ */
+
+namespace Extend\Integration\Model;
+
+use Extend\Integration\Service\Extend;
+use Extend\Integration\Api\ProductProtectionInterface;
+use Magento\Checkout\Model\Cart;
+use Magento\Catalog\Model\ProductFactory;
+
+class ProductProtection implements ProductProtectionInterface {
+  private Cart $cart;
+  private ProductFactory $productFactory;
+
+  public function __construct(
+    Cart $cart,
+    ProductFactory $productFactory
+  ) {
+      $this->cart = $cart;
+      $this->productFactory = $productFactory;
+  }
+    /**
+     * Add product protection to cart
+     *
+     * @param int $quantity
+     * @param string $productId
+     * @param string $planId
+     * @param int $price
+     * @param int $term
+     * @param string $coverageType
+     * @param string $token = null
+     * @param float $listPrice = null
+     * @param string $orderOfferPlanId = null
+     * @return void
+     */
+  public function add(int $quantity, string $productId, string $planId, int $price, int $term, string $coverageType, string $leadToken = null, float $listPrice = null, string $orderOfferPlanId = null): void {
+    $product = $this->productFactory->create();
+    $product->load($product->getIdBySku(Extend::WARRANTY_PRODUCT_SKU));
+
+    $params = array();
+    $options = array();
+    $params['qty'] = $quantity;
+    $params['product'] = $product->getId();
+    $product->setData('extend_plan_price', ($price/100));
+
+    foreach ($product->getOptions() as $o) {
+      $optionId = $o->getId();
+      switch($o->getTitle()) {
```

Comment:
I trust you but where does the reasoning for this logic come from? Possibly worth some commentary unless it's too elaborate, which I assume it is.